packages/app-core/src/ui/App/ViewContainer.tsx

```diff
@@ -56,6 +56,18 @@ const ViewContainer = observer(function ({
     }
   }, [ref, session, view])
 
+  useEffect(() => {
+    // ensures there is always one view in focus, if there is a view open
+    if (
+      session.focusedViewId &&
+      session.views.some(view => view.id === session.focusedViewId)
+    ) {
+      session.setFocusedViewId(session.focusedViewId)
+    } else {
+      session.setFocusedViewId(view.id)
+    }
+  }, [session.views.length, session, view.id])
```

Comment:
the useeffect might better be suited as an autorun

Reply:
I'm going to change this code block to

```
  useEffect(() => {
    if (!session.views.some(view => view.id === session.focusedViewId)) {
      session.setFocusedViewId(view.id)
    }
  }, [session.views.length, session, view.id])
```

I think that's less code than the auto run

What this block does is it sets the focused view id if a view is added or removed.

the if statement is necessary because if a view in focus is removed, we want to set the focused view id to the next view available.

i am also adding `e.target.getAttribute('data-testid') !== 'CloseIcon' && ..` to the `handleSelectView` function to prevent setting the focusedViewId if the user is closing the view -- this is relevant to prevent the change to the focusedViewId if the user is focused on a different view while closing another

let me know your thoughts

Reply:
I think that using an autorun is still worthwhile. it prevents subtle bugs from using "observable" stuff in the effect, which will only run in the case that the the effect "dependencies array" variables change. In this case, the code block lists dependencies [session.views.length, session, view.id] but the session.focusedViewId is not listed. therefore, there may be bugs related to this if session.focusedViewId changes, as the useEffect will not be 'alerted' to this update